src/js/providers/video-listener-mixin.js

```diff
@@ -170,11 +171,18 @@ const VideoListenerMixin = {
             3: 'Unknown decode error',
             4: 'File could not be played'
         }[code] || 'Unknown');
-        this.trigger(MEDIA_ERROR, {
-            code: code,
-            message: 'Error loading media: ' + message
-        });
+
+        this.trigger(
+            MEDIA_ERROR,
+            new PlayerError(`Error loading media ${message}`, HTML5_BASE_MEDIA_ERROR + Math.max(code, 0))
```

Comment:
@johnBartos the old codes don't map 1:1 to the new ones; `'Unknown network error'` is `221,000`, reason being it falls in the network category. The other codes are also different

Reply:
also `this.video.error` should be passed in as sourceError